[DonovanJJ] iP

DonovanJJ opened the pull request:
# notDuke
> Definitely not a Duke 👍 
**Helps to organise and keep track of Tasks to be done!**

## Here is how you can download it
1. Click on this link
1. Open the file
1. Use some of the commands shown below
1. Tweak the commands to your desire 
1. Give some love to notDuke ❤️ 

## Here are some examples of commands you could use:
- todo borrow book
- deadline return book /by 2/12/2019 1800
- event project meeting /from Mon 2pm /to 4pm
- list
- bye

## Features:
- [x] Managing of task list
- [ ] Voice support
- [ ] Upgraded GUI
- [ ] More support for suggested improvements!

If your keen on reading the code. Heres the main method:
```
public static void main(String[] args) {
    Duke bot = new Duke("./ipOutput.txt");
    bot.run();
    bot.ui.exit();
}
```

This could not have been done without the help of this [course](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

DonovanJJ committed:
Added: Rename, Greet, Exit

DonovanJJ committed:
Added: Echo

DonovanJJ committed:
Added: Add, List

DonovanJJ committed:
Added: Mark as Done

DonovanJJ committed:
Added: ToDos, Events, Deadlines

DonovanJJ committed:
Added: Handle Errors

DonovanJJ committed:
Added: Delete

DonovanJJ committed:
Added: TextUITesting

DonovanJJ committed:
Add Level-7 Save
Updated the code to read and write from an output file
This saves progress made by the user for the next session

DonovanJJ committed:
Merge branch 'branch-level-7'
Added Level-7

DonovanJJ committed:
Add support for date, time for deadlines
Updated the deadline class to support it

DonovanJJ committed:
Merge branch 'branch-level-8'
Add date, time format identification and display

DonovanJJ committed:
Redesign OOP
Ui, Storage, Parser, TaskList class have been added

DonovanJJ committed:
Merge branch 'A-MoreOOP'
Merge: A-MoreOOP

DonovanJJ committed:
Add packges duke and task
task - contains the different task classes, duke contains the main logic

DonovanJJ committed:
Merge branch 'A-Packages'
Merge A-Packages

DonovanJJ committed:
Merge branch 'add-gradle-support'
Merge add-support-gradle

DonovanJJ committed:
Add Junit for Deadlines and Parser method

DonovanJJ committed:
Merge tag 'A-Junit'
Merge Junit cases

DonovanJJ committed:
Add JavaDocs to all files for better documentation

DonovanJJ committed:
Updated code structure to adhere to coding standard

DonovanJJ committed:
Add minor changes to code to adhere to standards

DonovanJJ committed:
Add Find Functionality
Users can search for substrings of the task name

DonovanJJ committed:
Merge tag 'A-JavaDoc'
Merge Javadoc for better documentation of code

DonovanJJ committed:
Fix merge conflicts

DonovanJJ committed:
Resolved merge conflict with Find

DonovanJJ committed:
Update file path to use relative paths

DonovanJJ committed:
Update build gradle to remove bug in main file location

DonovanJJ committed:
Corrected the equals method for Deadline class

DonovanJJ committed:
Edit Code to comply with the code styling guidelines

DonovanJJ committed:
Add GUI with JavaFX

DonovanJJ committed:
Update gradle entry path

DonovanJJ committed:
Update Duke to only launch from the launcer. Removed the CLI version of the application

DonovanJJ committed:
Merge branch 'Level-10'
Merge with GUI branch

s-kybound commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
```
VERY well done, updating the README!

s-kybound commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+
+import java.io.IOException;
+import java.util.Collections;
+
```
Good job separating the imports into multiple different sections!

s-kybound commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ */
+public class DialogBox extends HBox {
```
Good job following class naming conventions!

s-kybound commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
```
Love the naming standards for methods, and making them sound like actions!

s-kybound commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+ */
+public class Duke {
+
+    private String outputPath;
```
Good names for the Duke components!

s-kybound commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String response = duke.getResponse(input);
+        dialogContainer.getChildren().addAll(
```
Good use of indentation to break up a long line into several!

s-kybound commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+        String[] arrStrings = input.split("\\s+");
+        String command = arrStrings[0];
+        String name = "";
+        if (command.equals("deadline")) {
```
Would it be better to refactor this into a switch case statement?

s-kybound commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+        String command = input.split("\\s+")[0];
+        String message = "";
+        int choice = -1;
+        switch (command) {
```
Good that you have followed the course's recommended switch case indentation level!

s-kybound commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,110 @@
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
```
Good use of Javadocs!

s-kybound commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,89 @@
+    private String from;
+    private String to;
+
+    /**
```
Excellent use of Javadocs tags and descriptions!

s-kybound commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,89 @@
+        this.from = from;
+    }
+
+    private String getTo() {
```
Good naming for getter functions!

s-kybound commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,87 @@
+     * @param word String repreentingt the name substring they are looking for
+     * @return {@code true} if word is a substring of the name
+     */
+    public Boolean containName(String word) {
```
Possibly better to rename?

```suggestion
    public Boolean containsName(String word) {
```

s-kybound commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,71 @@
+
+public class ParserTest {
+    @Test
+    public void createTaskTestTodoException() {
```
Would it be better to rename the test methods following the test methods conventions?

```
featureUnderTest_testScenario_expectedBehavior()
```

s-kybound reviewed (commented):
Overall, great naming, and style in general, convention! I can see the effort used to follow the course's guidance regarding this. However, to nitpick, there are minor improvements that can be explored.

DonovanJJ committed:
Updated coding standard based on feedback received

DonovanJJ committed:
Add Java Assertions

The code may not function correctly due to underlying assumptions such as return messages are non-empty and the parser always have initialised ui, storage, tasklist.

Utilised Java assertions to check for valid values

Ensures that the underlying assumptions always hold true

DonovanJJ committed:
Update Code Quality

Long methods were used and SLAP was not adhered to

Refractoring of code and regression testing was conducted to create more levels of abbstraction to simplify long methods

Readability of code is improved

DonovanJJ committed:
Merge pull request #2 from DonovanJJ/branch-A-CodeQuality

Update Code Quality

DonovanJJ committed:
Merged Codequality and assertions, fixed conflicts

DonovanJJ committed:
Merge pull request #1 from DonovanJJ/branch-A-Assertions

Add Java Assertions

DonovanJJ committed:
Add workflow CI

DonovanJJ committed:
Add duplicate task detection

Duplicate tasks were being accepted by the chat bot, this can cause anormally in behaviour

Created a new method that loops through the storage array list.
Checks if any task is equal to the added task"

DonovanJJ committed:
Add JUnit test for Event object creation

DonovanJJ committed:
Update README.md to contain a brief userguide

DonovanJJ committed:
Added Ui PNG file

DonovanJJ committed:
Merge branch 'Ui.png'

DonovanJJ committed:
Minor extraction of methods within Storage.

This helps to improve overall code quality and readability

DonovanJJ committed:
Resolve minor styling issues

DonovanJJ committed:
Fix minor bugs in docs/READme.md

DonovanJJ committed:
Update to docs/README.md

DonovanJJ committed:
Update docs/README.md

DonovanJJ committed:
Update docs/README.md

DonovanJJ committed:
Update docs/README.md

DonovanJJ committed:
Update minor bugs
